pkg/webhook/mutation/pod_mutator/pod_mutator.go

```diff
@@ -66,7 +66,14 @@ func (webhook *podMutatorWebhook) Handle(ctx context.Context, request admission.
 	emptyPatch := admission.Patched("")
 	mutationRequest, err := webhook.createMutationRequestBase(ctx, request)
 	if err != nil {
-		return silentErrorResponse(mutationRequest.Pod, err)
+		emptyPatch.Result.Message = fmt.Sprintf("unable to inject into pod (err=%s)", err.Error())
+		log.Error(err, "building mutation request base encountered an error")
+		span.RecordError(err)
+		return emptyPatch
+	}
+	if mutationRequest == nil {
```

Comment:
This is not a thing that can happen, if the `mutationRequest` is nil then we hit an error(so the if above will take over)

If we want to make sure, then write tests for it, but this is just dead code to me
```

Reply:
No, `createMutationRequestBase` has an path that explicitly does a `return nil, nil` in case of OLM stuff (https://github.com/Dynatrace/dynatrace-operator/blob/0202ee4c9736ab59e0a740f1072223f357b5d428/pkg/webhook/mutation/pod_mutator/request.go#L36)

Reply:
oh, its that OLM special scenario 🤦 
would it make sense to have a more specific message in this case ? (the current message may be spooky)

also a sidenote: you added the error to the `span` there, wouldn't it be possible to update/upgrade the `silentErrorResponse` to add the span also, so we could reuse it ? 🤔 (maybe out of scope)